pkg/http/http.go

```diff
@@ -95,10 +212,23 @@ func NewHTTPClient(cfg ClientConfig, name string) (*http.Client, error) {
 		return nil, err
 	}
 
-	client, err := config_util.NewClientFromConfig(httpClientConfig, name, config_util.WithHTTP2Disabled())
+	// see https://github.com/golang/go/issues/13801
+	if cfg.MaxIdleConns == 0 {
+		cfg.MaxIdleConns = 20000
+	}
+	if cfg.MaxIdleConnsPerHost == 0 {
+		cfg.MaxIdleConnsPerHost = 1000
+	}
+	rt, err := NewRoundTripperFromConfig(
+		httpClientConfig,
+		name,
+		WithMaxIdleConns(cfg.MaxIdleConns),
+		WithMaxIdleConnsPerHost(cfg.MaxIdleConnsPerHost))
 	if err != nil {
 		return nil, err
 	}
+	client := &http.Client{Transport: rt}
```

Comment:
Setting transport does not make sense, if we override it later on, why not passing `rt` just below?

Reply:
Makes sense